types/ua-parser-js/index.d.ts

```diff
@@ -139,7 +139,7 @@ declare namespace IUAParser {
 }
 
 declare module "ua-parser-js" {
-    class UAParser {
+    export class UAParser {
```

Comment:
This doesn't seem right to me.
In particular, the module _is_ the class (i.e. it's a constructable module) as shown in [the source](https://unpkg.com/ua-parser-js@0.7.19/src/ua-parser.js).
However, this change makes the type say that the module is an object with a property `UAParser`, which will fail at runtime.

If you want to use ES6 imports (i.e. `import UAParser from 'ua-parser-js';`), you can use the `--esModuleInterop` compiler setting. This doesn't seem right to me.
In particular, the module _is_ the class (i.e. it's a constructable module).
However, this change makes the type say that the module is an object with a property `UAParser`, which will fail at runtime.

If you want to use ES6 imports (i.e. `import UAParser from 'ua-parser-js';`), you can use the `--esModuleInterop` compiler setting.

Reply:
It doesn't fail at runtime though. There is this line in the source: `exports.UAParser = UAParser;`, so the module is the class, but also contains the class, which makes this syntax valid.

If you want, you can take a look at #25202 for more context on this PR.

Reply:
Oops, my mistake! I missed that piece.